tests/test_feedexport.py

```diff
@@ -464,6 +445,61 @@ def parse(self, response):
         data = yield self.run_and_export(TestSpider, settings)
         return data
 
+    @defer.inlineCallbacks
+    def assertExported(self, items, header, rows, settings=None, ordered=True):
+        yield self.assertExportedCsv(items, header, rows, settings, ordered)
+        yield self.assertExportedJsonLines(items, rows, settings)
+        yield self.assertExportedXml(items, rows, settings)
+        yield self.assertExportedPickle(items, rows, settings)
+        yield self.assertExportedMarshal(items, rows, settings)
+        yield self.assertExportedMultiple(items, rows, settings)
+
+    @abstractmethod
+    def run_and_export(self, spider_cls, settings):
+        pass
+
+    def _load_until_eof(self, data, load_func):
+        result = []
+        with tempfile.TemporaryFile() as temp:
+            temp.write(data)
+            temp.seek(0)
+            while True:
+                try:
+                    result.append(load_func(temp))
+                except EOFError:
+                    break
+        return result
+
+
+class FeedExportTest(FeedExportTestBase):
+    __test__ = True
+
+    @defer.inlineCallbacks
+    def run_and_export(self, spider_cls, settings):
+        """ Run spider with specified settings; return exported data. """
+
+        FEEDS = settings.get('FEEDS') or {}
+        settings['FEEDS'] = {
+            urljoin('file:', pathname2url(str(file_path))): feed
+            for file_path, feed in FEEDS.items()
+        }
+        content = {}
+        try:
+            with MockServer() as s:
```

Comment:
Maybe `scrapy.utils.test.get_crawler`?

Reply:
Do you mean to use `scrapy.utils.test.get_crawler` instead of `MockServer` or it is about an error which I get trying to use `S3FeedStorageTest::test_store`?

Reply:
Yeah, I don't know if you really need `MockServer`.
If it's required, then go for it.
Was just suggesting `get_crawler` as an alternative, in case you weren't aware of it :smile:

Reply:
Okay, finally got it.
MockServer was in FeedExportTest before, so believe there was any reason to do this such way. I will also remember that get_crawler is an alternative. =))